cpp/cmake/TemplateShellWrapper.cmake.in

```diff
@@ -2,7 +2,187 @@
+INSTALL_FULL_DATADIR="@CMAKE_INSTALL_FULL_DATADIR@"
+INSTALL_FULL_INFODIR="@CMAKE_INSTALL_FULL_INFODIR@"
+INSTALL_FULL_LOCALEDIR="@CMAKE_INSTALL_FULL_LOCALEDIR@"
+INSTALL_FULL_MANDIR="@CMAKE_INSTALL_FULL_MANDIR@"
+INSTALL_FULL_DOCDIR="@CMAKE_INSTALL_FULL_DOCDIR@"
+
+BF_INSTALL_DATADIR="@OME_BIOFORMATS_INSTALL_DATADIR@"
+BF_INSTALL_FULL_DATADIR="@OME_BIOFORMATS_INSTALL_FULL_DATADIR@"
+
+BF_INSTALL_ICONDIR="@OME_BIOFORMATS_INSTALL_ICONDIR@"
+BF_INSTALL_FULL_ICONDIR="@OME_BIOFORMATS_INSTALL_FULL_ICONDIR@"
+
+BF_INSTALL_LIBEXECDIR="@OME_BIOFORMATS_INSTALL_LIBEXECDIR@"
+BF_INSTALL_FULL_LIBEXECDIR="@OME_BIOFORMATS_INSTALL_FULL_LIBEXECDIR@"
+
+BF_INSTALL_SCHEMADIR="@OME_BIOFORMATS_INSTALL_SCHEMADIR@"
+BF_INSTALL_FULL_SCHEMADIR="@OME_BIOFORMATS_INSTALL_FULL_SCHEMADIR@"
+
+BF_INSTALL_TRANSFORMDIR="@OME_BIOFORMATS_INSTALL_TRANSFORMDIR@"
+BF_INSTALL_FULL_TRANSFORMDIR="@OME_BIOFORMATS_INSTALL_FULL_TRANSFORMDIR@"
+
+
+# Release version
+
+VERSION_MAJOR="@OME_VERSION_MAJOR@"
+VERSION_MINOR="@OME_VERSION_MINOR@"
+VERSION_PATCH="@OME_VERSION_PATCH@"
+VERSION_EXTRA="@OME_VERSION_EXTRA@"
+RELEASE_DATE="@OME_VCS_DATE_S@"
+
+# Program invoked as and command requested (if any)
+
+prog=$0
+if [ "$#" -gt 0 ]; then
+  cmd=$1
+  shift
+fi
+
+# Determine installation prefix if unset, then determine paths for
+# programs, libraries, manpages etc.
+
+if [ -d "$INSTALL_PREFIX" ]; then
+  PATH="$INSTALL_FULL_BINDIR:$PATH"
+  LIBEXECDIR="$BF_INSTALL_FULL_LIBEXECDIR"
+  MANDIR="$INSTALL_FULL_MANDIR"
+  if [ "$(uname -s)" = "Darwin" ]; then
+    if [ -z "$DYLD_LIBRARY_PATH" ]; then
+      export DYLD_LIBRARY_PATH="$INSTALL_FULL_LIBDIR"
+    else
+      export DYLD_LIBRARY_PATH="$INSTALL_FULL_LIBDIR:$DYLD_LIBRARY_PATH"
+    fi
+  else
+    if [ -z "$LD_LIBRARY_PATH" ]; then
+      export LD_LIBRARY_PATH="$INSTALL_FULL_LIBDIR"
+    else
+      export LD_LIBRARY_PATH="$INSTALL_FULL_LIBDIR:$LD_LIBRARY_PATH"
+    fi
+  fi
+else
+  bindir=$(dirname "$prog")
+  rootdir=${bindir%@CMAKE_INSTALL_FULL_BINDIR@}
+
+  PATH="${rootdir}${INSTALL_FULL_BINDIR}:$PATH"
+  LIBEXECDIR="${rootdir}${BF_INSTALL_FULL_LIBEXECDIR}"
+  MANDIR="${rootdir}${INSTALL_FULL_MANDIR}"
+  if [ "$(uname -s)" = "Darwin" ]; then
+    if [ -z "$DYLD_LIBRARY_PATH" ]; then
+      export DYLD_LIBRARY_PATH="$rootdir$INSTALL_FULL_LIBDIR"
+    else
+      export DYLD_LIBRARY_PATH="$rootdir$INSTALL_FULL_LIBDIR:$DYLD_LIBRARY_PATH"
+    fi
+  else
+    if [ -z "$LD_LIBRARY_PATH" ]; then
+      export LD_LIBRARY_PATH="$rootdir$INSTALL_FULL_LIBDIR"
+    else
+      export LD_LIBRARY_PATH="$rootdir$INSTALL_FULL_LIBDIR:$LD_LIBRARY_PATH"
+    fi
+  fi
+fi
+
+usage() {
+  cat <<EOF
+usage: $prog [--help] [--usage] <cmd> [<args>]
+
+available bf commands in '$LIBEXECDIR'
+
+EOF
+  ls -C "$LIBEXECDIR" | sed -e 's;^;	;'
+}
+
+version() {
+  cat <<EOF
+bf (OME Bio-Formats) $VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH$VERSION_EXTRA ($RELEASE_DATE)
+Copyright © 2006–2014 Open Microscopy Environment
```

Comment:
No, it'll need updating.  I'll do it by hand for this PR, but I'll look into automating it.